virtcontainers/container.go

```diff
@@ -609,12 +640,13 @@ func newContainer(sandbox *Sandbox, contConfig ContainerConfig) (*Container, err
 				return &Container{}, err
 			}
 
-			c.devices = append(c.devices, ContainerDevice{
+			storedDevices = append(storedDevices, ContainerDevice{
 				ID:            dev.DeviceID(),
 				ContainerPath: info.ContainerPath,
 			})
 		}
 	}
+	c.devices = filterDevices(sandbox, c, storedDevices)
```

Comment:
The filtering should not happen every time the `container` instance is created in memory. It only needs to happen at `create`, meaning this filtering function should only be called from the case `err != nil` here.